ngraph/core/reference/include/ngraph/runtime/reference/max_pool.hpp

```diff
@@ -110,6 +110,340 @@ void max_pool(const T* arg,
     }
     NGRAPH_SUPPRESS_DEPRECATED_END
 }
+
+namespace {
+void validate_max_pool_kernel_params(const size_t dims,
+                                     const Shape& kernel,
+                                     const Strides& kernel_strides,
+                                     const Strides& kernel_dilations,
+                                     const Shape& pads_begin,
+                                     const Shape& pads_end) {
+    NGRAPH_CHECK(kernel.size() == dims && kernel_strides.size() == dims && kernel_dilations.size() == dims &&
+                     pads_begin.size() == dims && pads_end.size() == dims,
+                 "One of the MaxPool params does not match the ",
+                 dims,
+                 "D implementation.\nkernel=",
+                 kernel,
+                 "\nkernel_strides=",
+                 kernel_strides,
+                 "\nkernel_dilations=",
+                 kernel_dilations,
+                 "\npads_begin=",
+                 pads_begin,
+                 "\npads_end=",
+                 pads_end);
+}
+
+template <typename T>
+struct Coord : public std::vector<T> {
+    Coord(const Shape& pads_begin) {
+        std::vector<T>::reserve(pads_begin.size());
+        for (const auto axis_padding : pads_begin) {
+            std::vector<T>::push_back(0 - axis_padding);
```

Comment:
I have to familiarize myself with this new `Coord` struct to understand the rest of the code, not sure I like it 🤓

Reply:
`Coord` is just like `Coordinate` except it can store different data types for the underlying dimensions. I need signed integers for kernel positions and `Coordinate` only allows for `size_t`.

I can make the `Coordinate` class a template with `size_t` as default and use it here instead if that makes sense.

Reply:
I've done an experiment and I can't make `Coordinate` a template class without huge amount of changes in the whole code base. I'd like to stick to this helper struct unless there are some very strong arguments to drop it.

Reply:
Please, add code comments so we will not loose this information.